crates/bevy_math/src/primitives/dim3.rs

```diff
@@ -214,6 +215,83 @@ impl InfinitePlane3d {
 
         (Self { normal }, translation)
     }
+
+    /// Computes the shortest distance between a plane transformed with the given `isometry` and a
+    /// `point`. The result is a signed value; it's positive if the point lies in the half-space
+    /// that the plane's normal vector points towards.
+    #[inline]
+    pub fn signed_distance(&self, isometry: Isometry3d, point: Vec3) -> f32 {
+        self.normal.dot(isometry.inverse() * point)
+    }
+
+    /// Injects the `point` into the plane transformed with the given `isometry`.
+    ///
+    /// This projects the point orthogonally along the shortest path onto the plane.
+    #[inline]
+    pub fn project_point(&self, isometry: Isometry3d, point: Vec3) -> Vec3 {
+        point - self.normal * self.signed_distance(isometry, point)
+    }
+
+    /// Computes the shortest distance between a plane going through `Vec3::ZERO` and a `point`.
+    /// The result is a signed value; it's positive if the point lies in the half-space
+    /// that the plane's normal vector points towards.
+    #[inline]
+    pub fn local_signed_distance(&self, point: Vec3) -> f32 {
+        self.normal.dot(point)
+    }
+
+    /// Injects the `point` into the plane going through `Vec3::ZERO`.
+    ///
+    /// This projects the point orthogonally along the shortest path onto the plane.
+    #[inline]
+    pub fn local_project_point(&self, point: Vec3) -> Vec3 {
+        point - self.normal * self.local_signed_distance(point)
+    }
+
+    /// Computes an [`Isometry3d`] which transforms points from the plane in 3D space with the given
+    /// `origin` to the XY-plane.
+    ///
+    /// The transformation is [congruent](https://en.wikipedia.org/wiki/Congruence_(geometry))
+    /// meaning it will preserve shapes. This is **not** the case with normal projections.
+    ///
+    /// See [`InfinitePlane3d::isometries_xy`] for more information.
+    #[inline]
+    pub fn isometry_into_xy(&self, origin: Vec3) -> Isometry3d {
```

Comment:
One thing that struck me is that these methods use a Vec3 offset, but other methods here use a full Isometry for positioning, what is the rationale? Is there API usecase for both versions of both groups of methods?

Reply:
You're right, we can generalize this as well. Now that I think about this, wouldn't it make sense then to just make the `InfinitePlane3d` always point to `Vec3::Z`?

Reply:
We could make the whole thing a UnitType. `::new` could be removed and `::from_points` could just spit out the `Isometry3D`